src/colvarcomp_gpath.cpp

```diff
@@ -153,8 +164,25 @@ void colvar::CartesianBasedPath::computeDistanceBetweenReferenceFrames(std::vect
     }
 }
 
-colvar::gspath::gspath(std::string const &conf): CartesianBasedPath(conf) {
+
+void colvar::CartesianBasedPath::apply_force(colvarvalue const &force)
+{
+    cvm::error("Error: using apply_force() in a component of type CartesianBasedPath, which is abstract.\n",
+               COLVARS_BUG_ERROR);
+}
+
+
+
+colvar::gspath::gspath()
+{
     set_function_type("gspath");
+}
+
+
+int colvar::gspath::init(std::string const &conf)
+{
+    int error_code = CartesianBasedPath::init(conf);
```

Comment:
Maybe we need to check the error code and return early too? If you think it is too complicated to return error code in derived classes, we can throw exceptions and catch them in some of the upper level functions.

Reply:
Dude, are you really advocating for supporting VMD and and propose adding exceptions in the same review round? ;-)

Reply:
Sorry, I should be clearer. Exception is just one of the solutions, and if I were you I might use exceptions in any `init` functions and catch them in `colvar::update_cvc_config`. Early return is another one, which may need more lines of code for both base and derived classes. Using either of them should be enough. As for VMD, it is just an example of interactive applications. Colvars as a library should not terminate the main interactive program if there is an error.

Reply:
I should be clearer, too :-) Exceptions would be convenient on their own, actually. We haven't used them in the past mostly to play it safe with VMD, which traditionally aims to support exotic and/or ancient compilers. Although compiler support has become more uniform with C++11, best practices like the [Google Style Guide](https://google.github.io/styleguide/cppguide.html#Exceptions) don't encourage vendors to support them well.

I think that adding early returns wherever it looks like the code may crash is probably the safest right now. I think that's still a minority of all error conditions.

Reply:
This is a common pattern in several places, I made similar "early return" changes in #650 and Haochuan added a bunch more in this PR.
I wouldn't mind having this in more places, that is, everywhere, and I wonder if there is a more concise / elegant / modern / systematic way of formulating this, besides exceptions. The C programmer in me would make it a macro... There, I said it, I'm old. C++11 best practices and Google Style Guide: get off my lawn.

Reply:
Just to be clear, I'm old too but I would be happy to embrace anything that the maintainers of a major package (whatever their age might be) will commit to approve.

*In writing, please.*